flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/RunLengthDecoder.java

```diff
@@ -265,8 +265,16 @@ void readNextGroup() {
                     while (valueIndex < this.currentCount) {
                         // values are bit packed 8 at a time, so reading bitWidth will always work
                         ByteBuffer buffer = in.slice(bitWidth);
-                        this.packer.unpack8Values(
-                                buffer, buffer.position(), this.currentBuffer, valueIndex);
+                        if (buffer.hasArray()) {
+                            this.packer.unpack8Values(
```

Comment:
I'm wondering why this changes will improve perfermance. IIUC, from the code
```java
public void unpack8Values(final byte[] input, final int inPos, final int[] output, final int outPos) {
    unpack8Values(ByteBuffer.wrap(input), inPos, output, outPos);
  }
```
 the method just delegates to `unpack8Values` and it needs extra `ByteBuffer.wrap(input)`.
Could you please explain a little bit?

Reply:
@luoyuxia This BytePacker is only abstract parent class, and unpack8Values will not run this method.
I have debug code, it run Packer7 class which is code generated.

Reply:
Thanks for explanation. Make sense to me.

Reply:
Could you please add some comments in here to expain why we change it?
Now, we use a deprecated method, we may need some comments for it so that the others can know the context when come to this method for the case that the method is deleted by a higher parquet version in some days and flink needs to upgrad to the  higher parquet version.

Reply:
@luoyuxia Good idea. I have added comments.  Besides, I have connected parquet-mr community to suggest remote the deprecated flag and change to use byte array default.